app/helpers/automated_tests_client_helper.rb

```diff
@@ -14,128 +14,97 @@ def create_test_repo(assignment)
     end
-
-    # Create/Update test support files (4 cases)
-    testsupporters.each do |file_num, file|
-      # 1) Remove an existing test support file
-      if testsupporters[file_num][:_destroy]
-        updated_support_files[file_num] = file.clone
-        next
+    # 4) Create new test file
+    if is_new
+      new_file_name = new_file.original_filename
+      if model_class.exists?(name_field => new_file_name, :assignment => assignment)
+        raise I18n.t('automated_tests.duplicate_filename') + new_file_name
       end
-      # 2) Empty form
-      next if new_support_file.nil? && testsupporters[file_num][:file_name].nil?
-      # 3) Create a new test support file
-      if testsupporters[file_num][:file_name].nil?
-        updated_support_files[file_num] = {}
-        filename = new_support_file.original_filename
-        if TestSupportFile.exists?(file_name: filename, assignment: assignment)
-          raise I18n.t('automated_tests.duplicate_filename') + filename
-        end
-        updated_support_files[file_num] = file.clone
-        updated_support_files[file_num][:file_name] = filename
-      # 4) Edit an existing test support file
-      else
-        # 4a) Modify test support file options only
-        if params[('new_update_file_' + testsupporters[file_num][:file_name]).to_sym].nil?
-          updated_support_files[file_num] = file.clone
-        # 4b) Replace test support file
-        else
-          new_update_file = params[('new_update_file_' + testsupporters[file_num][:file_name]).to_sym]
-          new_file_name = new_update_file.original_filename
-          old_file_name = file[:file_name]
-          if TestSupportFile.exists?(file_name: new_file_name, assignment: assignment)
-            raise I18n.t('automated_tests.duplicate_filename') + new_file_name
-          end
-          updated_support_files[file_num] = file.clone
-          updated_support_files[file_num][:file_name] = new_file_name
-          assignment_tests_path = File.join(
-                   MarkusConfigurator.markus_ate_client_dir,
-                   @assignment.repository_folder,
-                   new_file_name)
-          File.open(assignment_tests_path, 'w') { |f| f.write new_update_file.read }
-          old_file_path = File.join(MarkusConfigurator.markus_ate_client_dir, @assignment.repository_folder,
-                                    old_file_name)
-          if File.exist?(old_file_path)
-            File.delete(old_file_path)
-          end
-        end
+      updated_form_file[name_field] = new_file_name
+      new_file_path = File.join(MarkusConfigurator.markus_ate_client_dir, assignment.repository_folder, new_file_name)
+      files.push({path: new_file_path, upload: new_file})
+    # 5) Possibly replace existing test file
+    else
+      return updated_form_file unless form_file[name_field].nil? # replacing a test file resets the old name
+      old_file_name = model_class.where(id: form_file[:id]).pluck(name_field).first

```

Comment:
You should be able to use `find` here.

Reply:
Yes but I can't pluck that way, I have to get the whole row. It's efficiency vs readability, I was choosing the first.

Reply:
I prefer readability on this line. Please change it.